### DISCUSSION TITLE
No Caddyfile found

### DISCUSSION THREAD

**Discussion by corri57** [indomus/forum#3288] [GENERALE - Discussioni generiche]
Ciao,
ho configurato il mio Caddyfile nel folder /config/caddy, ma qualcosa non va.
Guardando i registri del componente Caddy 2 vedo che non trova Caddyfile in quanto se lo aspetta nel folder /etc/caddy quindi penso che utilizzi un non_caddyfile_config. Come posso fare per far utilizzare il mio Caddyfile?

**Comment by snakuzzo** [comment]
Da quello che leggo stai usando HA OS, quindi l'addon di Caddy 2 (che non è un componente).
Inoltre, devi capire che ogni addon è un container docker e quello che leggi nei messaggi è relativo al container, non a quello che c'è al di fuori.
Poi...non so dove tu abbia letto di mettere il file sotto `/config/caddy`, ma sappi che è sbagliato.
Nella [nostra guida](https://indomus.it/guide/installare-e-configurare-caddy-su-home-assistant-os-hassos-hassio/#instacaddy) (così come nella [documentazione ufficiale dell'addon](https://github.com/einschmidt/hassio-addons/blob/main/caddy-2/DOCS.md#add-on-configuration-directory)) è spiegato esattamente dove e come va copiato

Di seguito uno screenshot preso dal nostro articolo

![image](https://github.com/user-attachments/assets/86169d2e-75f0-44b5-a89d-1f57477c2edc)

Nello specifico caso dell'addon, sulla nostra guida tutto il giro per l'utilizzo del Caddyfile è fatto solamente per spiegare come funziona e...

- se non hai particolari esigenze per esporre più servizi con più subdomain
- se sei assolutamente sicuro di aver aperto correttamente le porte per le challenge http-01 (porta 80) o tls-alpn (porta 443) e quindi evitando il test con lo staging

...puoi semplicemente NON usare il Caddyfile e usare la configurazione di base dell'addon, opportunamente valorizzata.

Es.

![image](https://github.com/user-attachments/assets/7aaa6cf8-2250-4dde-b2be-69cf77305b95)

Ovviamente, ricordati anche la parte di configurazione finale nel `configuration.yaml` (prima di avviare l'addon)...


```yaml
http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 127.0.0.1
    - 192.168.1.0/24  # qui ovviamente metti la subnet della tua LAN che potrebbe essere diversa
    - ::1
    - 172.30.33.0/24
```

Io però mi domando... perché venite qui a fare sempre le stesse domande con sempre i soliti problemi ?
Prima di chiedere aiuto la fate una benedetta ricerca su google, sul sito, sul forum ???
Questa cosa è davvero assurda.
